src/EventStore.ClientAPI/GossipSeed.cs

```diff
@@ -24,26 +25,36 @@ public class GossipSeed: IGossipSeed {
 		/// The host header to be sent when requesting gossip.
 		/// </summary>
 		public readonly string HostHeader;
+		
+		private readonly bool _v20Compatibility;
 
 		/// <summary>
 		/// Creates a new <see cref="GossipSeed" />.
 		/// </summary>
 		/// <param name="endPoint">The <see cref="IPEndPoint"/> for the External HTTP endpoint of the gossip seed. The standard port is 2113.</param>
 		/// <param name="hostHeader">The host header to be sent when requesting gossip. Defaults to String.Empty</param>
 		/// <param name="seedOverTls">Specifies that eventstore should use https when connecting to gossip</param>
-		public GossipSeed(IPEndPoint endPoint, string hostHeader = "", bool seedOverTls = false) {
+		/// <param name="v20Compatibility">If v20 compatibility mode is enabled.</param>
+		public GossipSeed(IPEndPoint endPoint, string hostHeader = "", bool seedOverTls = false, bool v20Compatibility = false) {
 			EndPoint = endPoint;
 			HostHeader = hostHeader == string.Empty ? null : hostHeader;
 			SeedOverTls = seedOverTls;
+			_v20Compatibility = v20Compatibility;
 		}
 		
 		public string ToHttpUrl() {
+			if (_v20Compatibility && !string.IsNullOrEmpty(HostHeader)) {
```

Comment:
This change also appears to be unnecessary based on: https://github.com/EventStore/EventStore/pull/2860/files?diff=split&w=1#r600182774

Reply:
I disgree: https://github.com/EventStore/EventStore/pull/2860#discussion_r600227184